packages/widget/src/state/swapExecutionPage.ts

```diff
@@ -450,11 +450,11 @@ export const skipSubmitSwapExecutionAtom = atomWithMutation((get) => {
             if (getSigners?.getSvmSigner) {
               return getSigners.getSvmSigner();
             }
-            const walletName = wallets.svm?.walletName;
-            if (!walletName) throw new Error("getSvmSigner error: no svm wallet");
-            const solanaWallet = solanaWallets.find((w) => w.name === walletName);
-            if (!solanaWallet) throw new Error("getSvmSigner error: wallet not found");
-            return solanaWallet;
+            const adapter = await getSvmSigner();
+            if (!adapter) {
+              throw new Error("getSvmSigner error: no SVM wallet");
+            }
+            return adapter;
```

Comment:
@codingki why can't we handle this in here instead of passing it through as a prop?